From 029cecb45ceb652b0add04388fcaabe822e83660 Mon Sep 17 00:00:00 2001
From: Konstantin Khlebnikov <khlebnikov@openvz.org>
Date: Mon, 29 Dec 2008 19:37:47 +0300
Subject: [PATCH] cpt: Make the proper check for sigmask

invalid check of TS_RESTORE_SIGMASK (always false!)

original code ..rhel5..2.6.24 code from diff-cpt-sigsuspend-lockup-20070131
if (!signal_pending(current) &&
!test_thread_flag(TIF_RESTORE_SIGMASK)) {

TIF_RESTORE_SIGMASK replaced with TS_RESTORE_SIGMASK and
after commit 7648d96 setting TS_RESTORE_SIGMASK always set TIF_SIGPENDING.

so, second check is not needed.

http://bugzilla.openvz.org/show_bug.cgi?id=1122

Signed-off-by: Konstantin Khlebnikov <khlebnikov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---
 kernel/cpt/rst_process.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/kernel/cpt/rst_process.c b/kernel/cpt/rst_process.c
index f3f383c..a05390e 100644
--- a/kernel/cpt/rst_process.c
+++ b/kernel/cpt/rst_process.c
@@ -814,8 +814,7 @@ static void rst_restart_sys(void)
 		}
 	}
 
-	if (!signal_pending(current) &&
-	    !current_thread_info()->status & TS_RESTORE_SIGMASK) {
+	if (!signal_pending(current)) {
 		if (SYSCALL_ERRNO(regs) == ERESTARTSYS ||
 		    SYSCALL_ERRNO(regs) == ERESTARTNOINTR ||
 		    SYSCALL_ERRNO(regs) == ERESTARTNOHAND) {
-- 
1.6.0.6

